pepper-apis/src/main/java/org/broadinstitute/ddp/export/DataExporter.java

```diff
@@ -718,13 +724,26 @@ private String formatParticipantToStructuredJSON(
         }
 
         // Retrieving information to compute the dsm record
-        MedicalRecordService medicalRecordService = new MedicalRecordService(ConsentService.createInstance());
         DateValue diagnosisDate = medicalRecordService.getDateOfDiagnosis(handle, user.getId(), statusDto.getStudyId()).orElse(null);
         DateValue birthDate = medicalRecordService.getDateOfBirth(handle, user.getId(), statusDto.getStudyId()).orElse(null);
         MedicalRecordService.ParticipantConsents consents = medicalRecordService
                 .fetchBloodAndTissueConsents(handle, user.getId(), userGuid, statusDto.getStudyId(), studyGuid);
-        DsmComputedRecord dsmComputedRecord = new DsmComputedRecord(birthDate, diagnosisDate,
-                consents.hasConsentedToBloodDraw(), consents.hasConsentedToTissueSample(), pdfConfigRecords);
+
+        LocalDate dateOfMajority = null;
+        if (birthDate != null) {
+            LocalDate localBirthDate = birthDate.asLocalDate().orElse(null);
+            if (localBirthDate != null) {
+                dateOfMajority = participant.getAgeOfMajorityRule().getDateOfMajority(localBirthDate);
```

Comment:
What if we compute the date-of-majority all in the `DataExporter`? We can query for the study's policy and store it into the `StudyExtract` so we can use it here. (The nice thing about that is we will end up only querying for the policy once.)